### ISSUE TITLE
System.Net.WebException: The remote server returned an error: (400) Bad Request.

### ISSUE THREAD

**Issue by vasyugan** [cweiske/grauphel#31]
I have upgraded grauphel to 0.6.1 and yet I get the same "bad request" error. 

```
06.10.2015 16:59:53 [ERROR]: Caught exception. Message: The remote server returned an error: (400) Bad Request.
06.10.2015 16:59:53 [ERROR]: Stack trace for previous exception:   at System.Net.HttpWebRequest.CheckFinalStatus (System.Net.WebAsyncResult result) [0x00000] in <filename unknown>:0 
  at System.Net.HttpWebRequest.SetResponseData (System.Net.WebConnectionData data) [0x00000] in <filename unknown>:0 
06.10.2015 16:59:53 [ERROR]: Rest of stack trace for above exception:    at System.Environment.get_StackTrace()
   at Tomboy.WebSync.Api.OAuth.MakeWebRequest(RequestMethod method, System.String url, System.Collections.Generic.List`1 parameters, System.String postData)
   at Tomboy.WebSync.Api.OAuth.WebRequest(RequestMethod method, System.String url, System.String postData)
   at Tomboy.WebSync.Api.OAuth.Post(System.String uri, IDictionary`2 queryParameters, System.String postValue)
   at Tomboy.WebSync.Api.OAuth.GetAuthorizationUrl()
   at Tomboy.WebSync.WebSyncPreferencesWidget.OnAuthButtonClicked(System.Object sender, System.EventArgs args)
   at System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod , System.Object , System.Object[] , System.Exception ByRef )
   at System.Reflection.MonoMethod.Invoke(System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(System.Object obj, System.Object[] parameters)
   at System.Delegate.DynamicInvokeImpl(System.Object[] args)
   at System.MulticastDelegate.DynamicInvokeImpl(System.Object[] args)
   at System.Delegate.DynamicInvoke(System.Object[] args)
   at GLib.Signal.ClosureInvokedCB(System.Object o, GLib.ClosureInvokedArgs args)
   at GLib.SignalClosure.Invoke(GLib.ClosureInvokedArgs args)
   at GLib.SignalClosure.MarshalCallback(IntPtr raw_closure, IntPtr return_val, UInt32 n_param_vals, IntPtr param_values, IntPtr invocation_hint, IntPtr marshal_data)
   at Gtk.Application.gtk_main()
   at Gtk.Application.Run()
   at Tomboy.GnomeApplication.StartMainLoop()
   at Tomboy.Application.StartMainLoop()
   at Tomboy.Tomboy.StartTrayIcon()
   at Tomboy.Tomboy.Main(System.String[] args)
06.10.2015 16:59:53 [ERROR]: Failed to get auth URL from https://owncloud.****.org/index.php/apps/grauphel. Exception was: System.Net.WebException: The remote server returned an error: (400) Bad Request.
  at System.Net.HttpWebRequest.CheckFinalStatus (System.Net.WebAsyncResult result) [0x00000] in <filename unknown>:0 
  at System.Net.HttpWebRequest.SetResponseData (System.Net.WebConnectionData data) [0x00000] in <filename unknown>:0 
06.10.2015 17:00:01 [ERROR]: Caught exception. Message: The remote server returned an error: (400) Bad Request.
06.10.2015 17:00:01 [ERROR]: Stack trace for previous exception:   at System.Net.HttpWebRequest.CheckFinalStatus (System.Net.WebAsyncResult result) [0x00000] in <filename unknown>:0 
  at System.Net.HttpWebRequest.SetResponseData (System.Net.WebConnectionData data) [0x00000] in <filename unknown>:0 
06.10.2015 17:00:01 [ERROR]: Rest of stack trace for above exception:    at System.Environment.get_StackTrace()
   at Tomboy.WebSync.Api.OAuth.MakeWebRequest(RequestMethod method, System.String url, System.Collections.Generic.List`1 parameters, System.String postData)
   at Tomboy.WebSync.Api.OAuth.WebRequest(RequestMethod method, System.String url, System.String postData)
   at Tomboy.WebSync.Api.OAuth.Post(System.String uri, IDictionary`2 queryParameters, System.String postValue)
   at Tomboy.WebSync.Api.OAuth.GetAuthorizationUrl()
   at Tomboy.WebSync.WebSyncPreferencesWidget.OnAuthButtonClicked(System.Object sender, System.EventArgs args)
   at System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod , System.Object , System.Object[] , System.Exception ByRef )
   at System.Reflection.MonoMethod.Invoke(System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(System.Object obj, System.Object[] parameters)
   at System.Delegate.DynamicInvokeImpl(System.Object[] args)
   at System.MulticastDelegate.DynamicInvokeImpl(System.Object[] args)
   at System.Delegate.DynamicInvoke(System.Object[] args)
   at GLib.Signal.ClosureInvokedCB(System.Object o, GLib.ClosureInvokedArgs args)
   at GLib.SignalClosure.Invoke(GLib.ClosureInvokedArgs args)
   at GLib.SignalClosure.MarshalCallback(IntPtr raw_closure, IntPtr return_val, UInt32 n_param_vals, IntPtr param_values, IntPtr invocation_hint, IntPtr marshal_data)
   at Gtk.Application.gtk_main()
   at Gtk.Application.Run()
   at Tomboy.GnomeApplication.StartMainLoop()
   at Tomboy.Application.StartMainLoop()
   at Tomboy.Tomboy.StartTrayIcon()
   at Tomboy.Tomboy.Main(System.String[] args)
06.10.2015 17:00:01 [ERROR]: Failed to get auth URL from https://owncloud.*****.org/index.php/apps/grauphel. Exception was: System.Net.WebException: The remote server returned an error: (400) Bad Request.
  at System.Net.HttpWebRequest.CheckFinalStatus (System.Net.WebAsyncResult result) [0x00000] in <filename unknown>:0 
  at System.Net.HttpWebRequest.SetResponseData (System.Net.WebConnectionData data) [0x00000] in <filename unknown>:0
```

This is the kind of entry I see in the webserver log:

```
217.95.221.202 - - [06/Oct/2015:17:35:35 +0200] "GET /index.php/apps/grauphel/api/1.0 HTTP/1.1" 200 4462 "-" "-"
217.95.221.202 - - [06/Oct/2015:17:35:36 +0200] "POST /index.php/apps/grauphel/oauth/request_token HTTP/1.1" 400 1076 "-" "-"
```

I don't see anything in the owncloud log file.

**Comment by cweiske** [comment]
You're referring to bug #25. Seems you do have a different issue than what I fixed there.

You can now try to check which exact response comes from the server with wireshark, or you could give me a test account on your owncloud instance so I can check myself (tell me the owncloud URL and login data via email to cweiske+grauphel@cweiske.de).

**Comment by vasyugan** [comment]
Done. you should have received a notification from my owncloud server. Thank you!

**Comment by cweiske** [comment]
grauphel is not getting the oauth parameters:

```
oauth_problem=parameter_absent&oauth_parameters_absent=oauth_consumer_key%26oauth_signature%26oauth_signature_method%26oauth_nonce%26oauth_timestamp
```

But as in #25, the HTTP request has the parameters in the Authorization header.

@vasyugan: Please create a `.php` file in your owncloud directory (where `index.php` lives) with the following content:

```
<?php phpinfo(); ?>
```

Then mail me the URL/name of the file, cweiske+grauphel@cweiske.de

**Comment by vasyugan** [comment]
Sent today. Sorry, I overlooked your request

**Comment by cweiske** [comment]
`phpinfo();` shows that HTTP auth data arrive in `$_SERVER["HTTP_XAUTHORIZATION"]` - which grauphel already fixed in #20 and #25.

I have no idea why it does not work here. I don't know what to do here, sorry.

**Comment by cultcom** [comment]
I'm having the same problem on my server running php in fastcgi mode with suexec.

I could track it down so far, that an exception is thrown in controller/oauthcontroller.php line 258 with the message "Missing required parameters".
That line is a call to "$provider->checkOAuthRequest" and I really don't know what is missing there.
